app/services/select_outdated_uploads_strategy.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# A strategy class that selects uploads older than a given or configured date.
+# Used by the `DirectUploadsSweeper`
+class SelectOutdatedUploadsStrategy
+  # @param [String,Time,Date] before_date a string or date/time object before which uploads will be selected
+  # @return [ActiveRecord::Relation]
+  def self.select(before_date: default_before_date)
+    new(before_date: before_date).select
+  end
+
+  def self.default_before_date
+    Settings.sdr_api.sweep_uploads_older_than
+  end
+
+  # @param [String,Time,Date] before_date a string or date/time object before which uploads will be selected
+  def initialize(before_date: default_before_date)
+    @before_date = coerce_to_date_time(before_date)
+  end
+
+  # @return [ActiveRecord::Relation]
+  def select
+    active_record_class.where(
+      active_record_class.arel_table[:created_at].lt(
+        before_date
+      )
+    )
+  end
+
+  private
+
+  attr_reader :before_date
+
+  def coerce_to_date_time(date_time_representation)
+    # All Time and Date instances respond to `:strftime`
+    return date_time_representation if date_time_representation.respond_to?(:strftime)
+
+    # Parse to Time instance if not already passed a date/time-like object
+    Chronic.parse(date_time_representation)
+  end
+
+  def default_before_date
+    self.class.default_before_date
+  end
+
+  def active_record_class
+    @active_record_class ||= ActiveStorage::Blob
```

Comment:
why ivar?

Reply:
Not needed, will make the change you suggest.

Reply:
Done.